File: week1/2231.cpp
#include<iostream>
using namespace std;

int add_digit(int n) {
	int add = 0;

	for (int i = n; i > 0; i /= 10) {
		add += i % 10;
	}
	return add;
}
int main(void) {
	int N;
	cin >> N;

	for (int i = 1; i < N; i++)
	{
		if (i + add_digit(i) == N) {
			cout << i << endl;
			break;
		}
	}
	return 0;
}

File: week1/2231.py
def add_digit(i, hap):
    hap += i%10
    i = i//10
    if i > 0 :
        return add_digit(i, hap)
    else:
        return hap
def bunhae(N):
    for i in range(1, N):
        hap=0
        hap=add_digit(i, hap) + i
        if hap == N:
            return i
        else:
            return 0
N = int(input())
print(bunhae(N))
